xgboost_ray/main.py

```diff
@@ -23,22 +24,28 @@
     from ray import logger
     from ray.services import get_node_ip_address
     from ray.exceptions import RayActorError, RayTaskError
-    from ray.util.queue import Queue
     from ray.actor import ActorHandle
+    from xgboost_ray.util import Event, Queue
+
     RAY_INSTALLED = True
 except ImportError:
-    ray = None
-    logger = None
-    get_node_ip_address = None
-    Queue = None
-    ActorHandle = None
+    ray = logger = get_node_ip_address = Queue = Event = ActorHandle = None
     RAY_INSTALLED = False
 
 from xgboost_ray.tune import _try_add_tune_callback
 
 from xgboost_ray.matrix import RayDMatrix, combine_data, \
     RayDeviceQuantileDMatrix, RayDataIter, concat_dataframes
-from xgboost_ray.session import init_session, put_queue
+from xgboost_ray.session import init_session, put_queue, \
+    set_session_queue
+
+
+class RayXGBoostTrainingError(RuntimeError):
+    pass
+
+
+class RayXGBoostTrainingStopped(RuntimeError):
```

Comment:
Can you add a comment describing what errors are and the difference between them